src/param/le.rs

```
@@ -496,29 +494,24 @@ pub enum LeExtAdvDataStatus {
 
 impl LeExtAdvEventKind {
     pub fn data_status(&self) -> LeExtAdvDataStatus {
-        match (self.data_status_1(), self.data_status_2()) {
-            (false, false) => LeExtAdvDataStatus::Complete,
-            (false, true) => LeExtAdvDataStatus::IncompleteMoreExpected,
-            (true, false) => LeExtAdvDataStatus::IncompleteTruncated,
+        let data_status = self.0[0] >> 5 & 0x03;
+        match data_status {
+            0 => LeExtAdvDataStatus::Complete,
+            1 => LeExtAdvDataStatus::IncompleteMoreExpected,
+            2 => LeExtAdvDataStatus::IncompleteTruncated,
             _ => unreachable!(),
```

I don't think we should panic on unexpected data. A newer version of the spec might assign meaning to that value. I'd recommend adding a `Reserved` variant to the enum.

---

Ok, done. Currently reserved sets 0xff which seems to be the value Zephyr uses for ' invalid status'.

---

I mean 0b11 of course... :)